### PR TITLE
[espnet3-1] Add Data Organizer

### PR DESCRIPTION
**Pull request by Masao-Someki**
## What did you change?

1. Removed ESPnet-EZ, as we are transitioning to ESPnet-3.
2. Added a data organizer module along with accompanying unit tests.

---

## Why did you make this change?

The new data organizer is designed to support multi-dataset experiments. It allows users to easily add or remove datasets from their experiments, improving flexibility and modularity.

---

## Is your PR small enough?

Yes. Although the removal of ESPnet-EZ results in 14 files being deleted, only 4 new files are added.
The added code is well-documented, with comprehensive docstrings, which accounts for a relatively high number of lines.

---

## Additional Context

#6133

### CODE REVIEW

**Comment by Masao-Someki** [comment]
Here's details of unit test cases.

### 🔷 **1. Basic Integration**
Basic Integration of train/valid/test combinations

Test Function Name | Description
-- | --
test_data_organizer_init | Initializes DataOrganizer with train/valid/test
test_data_organizer_without_test | Verifies behavior without test section
test_data_organizer_test_only | Validates usage of test-only pipelines
test_data_organizer_train_valid_multiple | Ensures multiple train/valid datasets are supported
test_data_organizer_empty_train_valid_ok | Confirms train/valid can be empty lists

### 🔷 **2. Transform / Preprocessor Application**
Behavior of transform and preprocessor combinations 


Test Function Name | Description
-- | --
test_combined_dataset | Combines datasets and applies transform
test_data_organizer_transform_only | Applies only transform to data
test_data_organizer_preprocessor_only | Applies only preprocessor to data
test_data_organizer_transform_and_preprocessor | Applies both transform and preprocessor
test_espnet_preprocessor_without_transform | Uses only ESPnet-style preprocessor (UID-based)
test_espnet_preprocessor_with_transform | Combines transform with ESPnet preprocessor


### 🔷 **3. Test Set Variants**
Tests for variations in test dataset configurations 

Test Function Name | Description
-- | --
test_data_organizer_test_multiple_sets | Handles multiple named test sets


### 🔶 **Error Cases**

| Test Function Name                              | Description                                                  | Expected Exception |
| ----------------------------------------------- | ------------------------------------------------------------ | ------------------ |
| `test_data_organizer_train_only_assertion`      | Raises error when only train is provided without valid       | `RuntimeError`     |
| `test_data_organizer_inconsistent_keys`         | Fails when dataset output keys are inconsistent in a combined setup | `AssertionError`   |
| `test_data_organizer_transform_none`            | Simulates transform failure that raises an internal exception | `ValueError`       |
| `test_data_organizer_invalid_preprocessor_type` | Fails when a non-callable object is used as preprocessor     | `AssertionError`   |

**Comment by Masao-Someki** [comment]
The current CI error is mainly caused by the pyproject.toml configuration.
Since formatting-related changes are being handled separately in #6164 , the pyproject.toml in the base branch is not yet compatible with the current CI setup.

This PR should be merged only after #6139 has been merged and its CI has passed.

**Comment by sw005320** [comment]
This pull request introduces support for ESPnet-3 across multiple CI workflows and adds new functionality for dataset organization in the `espnet3` module. Key changes include updates to CI configurations, the addition of new Python scripts, and the introduction of a `DataOrganizer` class for managing datasets.

### CI Workflow Updates:
* [`.github/workflows/ci_on_debian11.yml`](diffhunk://#diff-b13dff138f65e577647f126f573d36d77fe805c8018d819d439da401f913456cR7-R14): Added support for the `espnet3` branch and updated job names and scripts to include ESPnet-3 testing. [[1]](diffhunk://#diff-b13dff138f65e577647f126f573d36d77fe805c8018d819d439da401f913456cR7-R14) [[2]](diffhunk://#diff-b13dff138f65e577647f126f573d36d77fe805c8018d819d439da401f913456cR47-R52)
* [`.github/workflows/ci_on_macos.yml`](diffhunk://#diff-e9868fff133d797c4c6a9e2ad3cbc3fc55bfd09b1a2cc1324218b5609567475fR7-R11): Added support for the `espnet3` branch.
* [`.github/workflows/ci_on_ubuntu.yml`](diffhunk://#diff-48d19e7512e7e9ec72c5503d3178348ad59275991f8b8334269329766e343de9R7-R11): Updated CI jobs to include ESPnet-3 testing, expanded Python and PyTorch versions for matrix testing, and added a step to fetch PR labels. [[1]](diffhunk://#diff-48d19e7512e7e9ec72c5503d3178348ad59275991f8b8334269329766e343de9R7-R11) [[2]](diffhunk://#diff-48d19e7512e7e9ec72c5503d3178348ad59275991f8b8334269329766e343de9L23-R25) [[3]](diffhunk://#diff-48d19e7512e7e9ec72c5503d3178348ad59275991f8b8334269329766e343de9L168-R170) [[4]](diffhunk://#diff-48d19e7512e7e9ec72c5503d3178348ad59275991f8b8334269329766e343de9L178-R181) [[5]](diffhunk://#diff-48d19e7512e7e9ec72c5503d3178348ad59275991f8b8334269329766e343de9R198-R200)
* [`.github/workflows/ci_on_windows.yml`](diffhunk://#diff-07e2a7a53f167a4bfe47fd72cc304f5071279a86c297985caaabd11ff35da7b1R7-R11): Added support for the `espnet3` branch.

### ESPnet-3 Module Enhancements:
* [`ci/test_python_espnet3.sh`](diffhunk://#diff-c35bd8cb103cc461d2a223e9296562eb72721f83f578c145eb338349b326d6d4L13-R16): Renamed from `ci/test_python_espnetez.sh` and updated to include ESPnet-3-specific paths and testing configurations.
* [`espnet3/data/__init__.py`](diffhunk://#diff-95851550ab39f5d69422356441d617837e9d578e242941a71f3d38a096c17bedR1-R2): Added imports for `data_organizer` and `dataset`.
* [`espnet3/data/data_organizer.py`](diffhunk://#diff-294170c415c4d9ca737ae8e58c78c357365c6b5e24418a4fb925889895fdc229R1-R187): Introduced the `DataOrganizer` class to manage training, validation, and test datasets, with support for modular dataset configuration and preprocessing.

**Comment by sw005320** [comment]
@Masao-Someki, can you fix the conflict?

File: .github/workflows/ci_on_ubuntu.yml [hunk 1]
```diff
@@ -177,16 +176,16 @@ jobs:
       max-parallel: 20
       matrix:
         os: [ubuntu-latest]
-        python-version: ["3.10"]
-        pytorch-version: ["2.6.0"]
-        chainer-version: ["6.0.0"]
+        python-version: ["3.10", "3.11"]
+        pytorch-version: [2.1.2, 2.2.2, 2.3.1, 2.4.1, 2.5.1, 2.6.0]
+        chainer-version: [6.0.0]
```
Comment (sw005320) [inline .github/workflows/ci_on_ubuntu.yml:181]: Do we need it?

Comment (Masao-Someki) [reply .github/workflows/ci_on_ubuntu.yml:181]: I thought it would be good to include those versions in the tests for ESPnet-3 as well, since we already test them for ESPnet1/2 and have table for them in the [Readme](https://github.com/espnet/espnet/tree/espnet3?tab=readme-ov-file#espnet-end-to-end-speech-processing-toolkit).

Comment (sw005320) [reply .github/workflows/ci_on_ubuntu.yml:181]: Not a version, but a chainer
Do you need chainer?

Comment (Masao-Someki) [reply .github/workflows/ci_on_ubuntu.yml:181]: Thank you, I understand now. Chainer is required by espnet-1 and is supposed to be removed in #6179.
However, it looks like the PR didn’t actually remove it, so I’ll go ahead and remove the Chainer version in that PR.

File: .github/workflows/ci_on_ubuntu.yml [hunk 2]
```diff
@@ -225,9 +238,9 @@ jobs:
       max-parallel: 20
       matrix:
         os: [ubuntu-latest]
-        python-version: ["3.10", "3.11"]
-        pytorch-version: [2.1.2, 2.2.2, 2.3.1, 2.4.1, 2.5.1, 2.6.0]
-        chainer-version: [6.0.0]
+        python-version: ["3.10"]
+        pytorch-version: ["2.6.0"]
+        chainer-version: ["6.0.0"]
```
Comment (sw005320) [inline .github/workflows/ci_on_ubuntu.yml:243]: Can we keep chainer here?

**Comment by sw005320** [comment]
Please fix https://github.com/espnet/espnet/actions/runs/16596452146/job/46944176786?pr=6167#step:8:302

**Review by sw005320 — APPROVED** [review]

**Comment by sw005320** [comment]
It looks good to me.
I'll merge after CI.
Please work on the next PRs.

**Comment by Masao-Someki** [comment]
Could you merge or rerun the CI? It seems one integration test fails with timeout

> FAILED test/espnet2/bin/test_asr_inference.py::test_Speech2Text_hugging_face_causal_lm[--akreal/tiny-random-BloomForCausalLM] - Failed: Timeout >30.0s
